### PR TITLE
add crc32 algorithm with few tests

### PR DESCRIPTION
**Pull request by CormickKneey**
## Description
Implemented the digest crc32 algorithm with some ut

### CODE REVIEW

File: dragonfly-client-util/src/digest/mod.rs [hunk 1]
```diff
@@ -131,7 +132,14 @@ pub fn calculate_file_hash(algorithm: Algorithm, path: &Path) -> ClientResult<Di
     let f = std::fs::File::open(path)?;
     let mut reader = std::io::BufReader::new(f);
     match algorithm {
-        Algorithm::Crc32 => Err(ClientError::Unimplemented),
+        Algorithm::Crc32 => {
+            let mut hasher = crc32fast::Hasher::new();
+            let mut buffer = Vec::new();
```
Comment (gaius-qi) [inline dragonfly-client-util/src/digest/mod.rs:137]: Need to set the buffer size and read the buffer of the file in a loop to update the hash.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Comment by gaius-qi** [comment]
@CormickKneey Fix lint and sign commits refer to https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification.

**Comment by CormickKneey** [comment]
PTL,bro! @gaius-qi

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**